qupath-core/src/main/java/qupath/lib/analysis/images/ContourTracing.java

```diff
@@ -588,38 +591,63 @@ public static Map<Number, ROI> createROIs(Raster raster, int band, RegionRequest
 	 * @return an ordered map containing all the ROIs that could be found; corresponding labels are keys in the map
 	 */
 	public static Map<Number, ROI> createROIs(SimpleImage image, RegionRequest region, int minLabel, int maxLabel) {
-		// Check how many labels are needed
-		float[] pixels = SimpleImages.getPixels(image, true);
-		if (maxLabel < minLabel) {
-			float maxValue = minLabel;
-			for (float p : pixels) {
-				if (p > maxValue)
-					maxValue = p;
+		var envelopes = new HashMap<Number, Envelope>();
+		if (minLabel != maxLabel) {
```

Comment:
Can we invert this if/else? Keeps the start of the two branches closer to the relevant condition

Reply:
I don't mind. I thought it was easier to read to show the use of envelopes before showing their non-use in a special case.